src/analysis.cpp

```diff
@@ -1322,6 +1326,192 @@ PolymerShape::PolymerShape(const json& j, Space& spc) : Analysisbase(spc, "Polym
     }
 }
 
+void SASAAnalysis::_to_disk() {
+    if (auto stream = std::ofstream(filename + "_histogram.dat")) {
+        stream << sasa_histogram;
+    }
+    if (output_stream) {
+        output_stream->flush();
+    }
+}
+
+void SASAAnalysis::_from_json(const json& json_input) {
+    probe_radius = json_input.value("radius", 1.4) * 1.0_angstrom;
+    slices_per_atom = json_input.value("slices", 20);
+    sasa_histogram.setResolution(json_input.value("resolution", 50.0_angstrom), 0.);
+
+    if (filename = json_input.value("file", ""s); !filename.empty()) {
+        output_stream = IO::openCompressedOutputStream(MPI::prefix + filename);
+        *output_stream << "# step SASA SASA" + u8::squared + " \n";
```

Comment:
I don't think you write sasa squared? (and it's not needed). Remove white space at end of line.